packages/e2e-tests/specs/editor/various/preview.test.js

```diff
@@ -304,3 +306,42 @@ describe( 'Preview with Custom Fields enabled', () => {
 		await editorPage.bringToFront();
 	} );
 } );
+
+describe( 'Preview with private custom post type', () => {
+	beforeAll( async () => {
+		await activatePlugin( 'gutenberg-test-custom-post-types' );
+	} );
+
+	afterAll( async () => {
+		await deactivatePlugin( 'gutenberg-test-custom-post-types' );
+	} );
+
+	it( 'should not show the Preview Externally button', async () => {
+		await createNewPost( { postType: 'not_public' } );
+
+		// Type in the title filed.
+		await page.type( '.editor-post-title__input', 'aaaaa' );
+		await page.keyboard.press( 'Tab' );
+
+		// Open the preview menu.
+		await page.click( '.editor-post-preview__button-toggle' );
+
+		// Expect there to be a single menu group in the dropdown.
+		const menuGroups = await page.$$(
+			'.editor-post-preview__dropdown-content .components-menu-group'
+		);
+		expect( menuGroups.length ).toBe( 1 );
+
+		// Expect there to be three preview buttons.
+		const previewButtons = await page.$$(
+			'.editor-post-preview__button-resize'
+		);
+		expect( previewButtons.length ).toBe( 3 );
```

Comment:
I'm wondering if we really need to check how many menu groups and resize buttons display in this test. Those things could change in the future, and this test shouldn't fail unless "Preview Externally" is actually showing.

Reply:
I've changed it so that it now only checks for the link.

The reason I wrote it like that originally is that I wanted to be able to check for regressions of this bug, and the bug is that the menu group and icon are displayed. I wouldn't be able to add assertions for the menu group or icon without adding some kind of unique identifier to them.

Reply:
Oh, I see. I wonder if a snapshot test would be a better way of checking for that scenario. If we did change the markup, we'd be expecting snaphsots to fail anyway.

Reply:
I personally feel like this is fine as it is. The test wouldn't be perfect, but it'd catch the most important failure, the presence of the link.

Snapshots for individual pieces of UI in end to end tests is a little outside of the norm, and while I'm not against it, it'd mean the snapshot needed to be updated for any minor unrelated change. That seems like it'd be a slower/more regular process than updating the element counts that were here previously.

Reply:
No worries! 
I was thinking a snapshot as part of the unit tests, but that doesn't need to be part of this changeset.